Transactions from a bakery Project.

In this project we analyze The dataset consists of 21293 observations from a bakery. The data file contains four variables, Date, Time, Transaction ID and Item. we not general data structure, summary statistics and we discover relationship between different features.
for more information about the dataset follow this link: https://www.kaggle.com/datasets/sulmansarwar/transactions-from-a-bakery?resource=download


First we import our needed libraries Pandas and numpy. ect....

In [75]:
import pandas as pd
import plotly.express as px
import numpy as np

We first Import our data set using pandas function read from CSV

In [39]:
df = pd.read_csv('BreadBasket_DMS.csv') 

Now lets have a look at our dataset heading the first 5 rows

In [41]:
print(df.head())

         Date      Time  Transaction           Item
0  2016-10-30  09:58:11            1          Bread
1  2016-10-30  10:05:34            2   Scandinavian
2  2016-10-30  10:05:34            2   Scandinavian
3  2016-10-30  10:07:57            3  Hot chocolate
4  2016-10-30  10:07:57            3            Jam


Let's look at summary statistics and counts for catagorical featuers

In [52]:
print(df[['Transaction','Item']].groupby('Item').count().dropna(),df.describe())

                          Transaction
Item                                 
Adjustment                          1
Afternoon with the baker           44
Alfajores                         369
Argentina Night                     7
Art Tray                           38
...                               ...
Tshirt                             21
Valentine's card                   13
Vegan Feast                        16
Vegan mincepie                     54
Victorian Sponge                    7

[95 rows x 1 columns]         Transaction
count  21293.000000
mean    4951.990889
std     2787.758400
min        1.000000
25%     2548.000000
50%     5067.000000
75%     7329.000000
max     9684.000000


We noticed that there is some duplicates in our dataframe, so let's find them. Using attribute duplicates we can easily get this info and then
we subset it into new dataframe.

In [64]:
df_dup= df[df.duplicated()]
print(df_dup)

             Date      Time  Transaction          Item
2      2016-10-30  10:05:34            2  Scandinavian
25     2016-10-30  10:27:21           11         Bread
39     2016-10-30  10:34:36           15          NONE
52     2016-10-30  10:49:29           21        Coffee
107    2016-10-30  12:23:01           49        Coffee
...           ...       ...          ...           ...
21173  2017-04-08  15:04:45         9618          Cake
21187  2017-04-08  16:05:07         9627        Coffee
21206  2017-04-08  16:30:32         9634        Coffee
21248  2017-04-09  11:40:10         9664        Coffee
21258  2017-04-09  12:04:13         9667      Sandwich

[1653 rows x 4 columns]


As we can see the duplicates happen to be 1653 times in our dataset, but which item had the most duplicates, let's discover them.

In [73]:
df_dup_itm=df_dup[['Transaction','Item']].groupby('Item').count().dropna()
print(df_dup_itm.sort_values(by='Transaction'))

                          Transaction
Item                                 
Afternoon with the baker            1
Tartine                             1
Chocolates                          1
Truffles                            1
Coke                                1
Scandinavian                        2
Mineral water                       2
Vegan mincepie                      2
Farm House                          3
Juice                               4
Muffin                              6
Hearty & Seasonal                   7
Jam                                 7
Soup                               16
Fudge                              17
Cookies                            25
Alfajores                          25
Medialuna                          31
NONE                               33
Hot chocolate                      38
Pastry                             41
Cake                               42
Tea                                85
Sandwich                           91
Bread       

To better understand now we use plotly to see how thoese duplicated values look like.

In [80]:
fig=px.bar(df_dup_itm,x='Transaction',labels={'Transaction':'Duplicated Transaction'})
fig.show()

So, the graph above shows us that most of the duplicated transactions are comming from coffee, Bread, Tea and Sandwich Which leave us under a question that small does small Items are most probablly tend to be scaned twice?